app/controllers/concerns/course/forum/topic_controller_subscription_concern.rb

```diff
@@ -0,0 +1,37 @@
+module Course::Forum::TopicControllerSubscriptionConcern
+  extend ActiveSupport::Concern
+
+  def subscribe
+    if set_subscription_state
```

Comment:
I also think it should be the case. But _how_ you intend to do this would dictate whether using the subscribe permission is an acceptable solution. If you need to declare two rules, then you ought to check for both here in case someone forgets to check for show together with subscribe.